### PR TITLE
Iss 6 dashboard skeleton

### LINKED ISSUE #6: FE: Dashboard Skeleton
- Create the skeleton for a logged in user experience - has a navbar + content pages (no content needed yet, this is just a skeleton, but the navbar should be fully functional)
- One navbar used throughout the whole logged in user experience (navbar is properly styled and shows a placeholder image for the user)
- The dropdown for the profile in the navbar should just show two links - logout (add logout functionality), + go to profile (the home page for a logged in user)
- click on the navbar and move between different pages (those pages can be blank for now)
- You have a parent component that holds all information for a logged in user 
- Part of the goal of this ticket is to have the whole user information accessible across all pages when the user is logged in - higher order components, React context or just a parent component where you pass information below
- To get user information, you *will need to make an API call* every time you open the app, to correctly get up to date logged in user information 
- This means you may need to make a "get user route" - that just gets the details of the logged in user
    - We don't want to store user details in JWT, Why?

### PR DESCRIPTION
**Pull request by Brian-Fairbanks**
Closes #26 

`✅ Create the skeleton for a logged in user experience - has a navbar + content pages (no content needed yet, this is just a skeleton, but the navbar should be fully functional)`

Skeleton has been set up, confirmed by details below

`✅ One navbar used throughout the whole logged in user experience (navbar is properly styled and shows a placeholder image for the user)`

the NavBar component is set up in app.js, enclosed in the router so other content pages can be loaded. (placeholder is visible when inverting the login check on line 78 of NavBar component)

```
        {!user ? (
```
so that login will be displayed when no user is found, and the valid username/avatar will be displayed when logged in.

`✅ The dropdown for the profile in the navbar should just show two links - logout (add logout functionality), + go to profile (the home page for a logged in user)`

Dropdown (MaterialUI's Menu component) shows Profile and Logout.  See attached image. 

`✅ click on the navbar and move between different pages (those pages can be blank for now)`

explore, launch, login, and profile pages can be swapped between.  The nav bar updates the active page accordingly.

`✅ You have a parent component that holds all information for a logged in user `

I have set up Components/User/UserContext, and implemented a context provider, so that any component can access the user data.  An example can be found in the navbar component

`✅ To get user information, you will need to make an API call every time you open the app, to correctly get up to date logged in user information`

I am running Components/User/getUserInfo on refresh to try and get new user information.  This is currently returning 404, thus user is not signed on.

`☐ This means you may need to make a "get user route" - that just gets the details of the logged in user`

very true, but this will require assistance with routes & authentication on the backend.  This should be addressed in future tickets.

`? We don't want to store user details in JWT, Why?`

The JWT should be proof of authentication, not a local storage for user data.  Storing any information in the cookies would decrease the security.  It would be better to have the server load and store this information to its state only after the bearer of a JWT provides a "proof of valid sign in" token.

Image of fluid navbar, on the launch page, while displaying dropdown (username has been clicked)
image

Image of nav bar while using is not signed in.
image

### CODE REVIEW

File: client/src/components/NavBar/index.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      <Box display="flex" id="nav-Left">
+        <img src={logo} className={classes.logo} alt="Logo" />
+        <Typography variant="h6">Product Launch</Typography>
+      </Box>
+
+      {/* Links and login */}
+      <Box display="flex" alignItems="center" id="nav-Right">
+        <NavLink
+          className={classes.link}
+          to="explore"
```
Comment (dproc96) [inline client/src/components/NavBar/index.js]:
```suggestion
          to="/explore"
```
While this does work as is, it would be better practice to include the slash

File: client/src/components/NavBar/index.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+              alignItems="center"
+              className={classes.link}
+              aria-controls="userDropdown"
+              aria-haspopup="true"
+              onClick={handleClick}
+            >
+              <Typography variant="subtitle1">
+                {user ? user.username : "Test Name"}
+              </Typography>
+              <Avatar></Avatar>
```
Comment (dproc96) [inline client/src/components/NavBar/index.js]: This content is appearing very close together on my screen, can you add some space between them?

File: client/src/components/User/getUserData/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          const data = await res.json();
+          // console.log(data.user);
+          setUser(data.user);
+          setLoading(false);
+        })
+        .catch((err) => {
+          setLoading(false);
+        });
+    }
+    findUser();
```
Comment (dproc96) [inline client/src/components/User/getUserData/index.js]: Is this feature in scope for this issue? Right now there's no way to test it as the backend hasn't been set up to handle that route

File: client/src/components/User/getUserData/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from "react";
```
Comment (dproc96) [inline client/src/components/User/getUserData/index.js]: Since this is a custom hook instead of a component, it would make sense to move this to a hooks folder instead of the components folder

File: client/src/App.js [hunk 5]
```diff
@@ -2,17 +2,33 @@ import React from "react";
 import { MuiThemeProvider } from "@material-ui/core";
 import { BrowserRouter, Route } from "react-router-dom";
 
-import { theme } from "./themes/theme";
-import LandingPage from "./pages/Landing";
+// User Context Information
+import { UserContext } from "./components/User/UserContext";
+import getUserData from "./components/User/getUserData";
```
Comment (dproc96) [inline client/src/App.js]: For clarity, it would be best to keep this name consistent with the name in the file `useFindUser`

File: client/src/components/User/getUserData/index.js (hunk 4 above)
Comment (dproc96) [reply client/src/components/User/getUserData/index.js]: It would also be good to align the names of the file and the default export. I recommend having the name start with `use` as it's a custom hook

File: client/src/themes/theme.js [hunk 6]
```diff
@@ -2,13 +2,20 @@ import { createMuiTheme } from "@material-ui/core";
 
 export const theme = createMuiTheme({
   typography: {
-    fontFamily: '"Roboto"',
+    fontFamily: ["Proxima Nova", "Roboto"].join(","),
```
Comment (dproc96) [inline client/src/themes/theme.js]:
```suggestion
    fontFamily: "'Proxima Nova', 'Roboto'"
```
This join operator isn't really necessary as the list of fonts is not created programmatically. As long as it's hardcoded like this you can just use a string

File: client/src/components/User/UserContext/index.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import { createContext } from "react";
```
Comment (dproc96) [inline client/src/components/User/UserContext/index.js]: Would be a good idea to move this file to a context folder rather than the components folder for clarity

**Review by dproc96 — COMMENTED** [review]
UI looks good, just had some minor comments and change requests that you can see below.

File: client/src/components/NavBar/index.js (hunk 1 above)
Comment (Brian-Fairbanks) [reply client/src/components/NavBar/index.js]: oops, Thank you Dan.  Changed, and will push shortly.

File: client/src/components/User/getUserData/index.js (hunk 3 above)
Comment (Brian-Fairbanks) [reply client/src/components/User/getUserData/index.js]: It might be a bit of a stretch, but it was specified in the ticket:

`To get user information, you will need to make an API call every time you open the app, to correctly get up to date logged in user information`

I figured I should set up the api call, expect 404 for now, and ensure it integrates properly later.

File: client/src/App.js (hunk 5 above)
Comment (Brian-Fairbanks) [reply client/src/App.js]: Oops.  Remnant of changing the name.  getUserData was not recognized as a valid hook anyway, so it has been renamed in all places to "useGetUserData".  Push will come shortly.

File: client/src/themes/theme.js (hunk 6 above)
Comment (Brian-Fairbanks) [reply client/src/themes/theme.js]: the join was used for legibility in the code,

``` 
fontFamily: [
      "Proxima Nova",
      "Roboto"
    ].join(","),
```

but it seems prettier liked a single line.  Unless the list grows, it probably isn't necessary though...

File: client/src/App.js [hunk 8]
```diff
@@ -2,17 +2,33 @@ import React from "react";
 import { MuiThemeProvider } from "@material-ui/core";
 import { BrowserRouter, Route } from "react-router-dom";
 
-import { theme } from "./themes/theme";
-import LandingPage from "./pages/Landing";
+// User Context Information
+import { UserContext } from "./context/UserContext";
+import useGetUserData from "./hook/useGetUserData";
```
Comment (tdellard1) [inline client/src/App.js]: I ran into some issues running this because there was a folder called hooks, but this is looking for a folder called hook (without the S)

Comment (Brian-Fairbanks) [reply client/src/App.js]: sorry about that.  Windows does not see the difference between Hooks and hooks (but unix systems do), so my fix to was to rename it "hook" in the interim, but then got a bit distracted between the pushes.  Sorry about the error. It should be fixed now.

File: client/src/components/NavBar/index.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+  },
+  subLink: {
+    textDecoration: "none",
+    padding: 1,
+    color: theme.palette.common.black,
+  },
+  nav: {
+    display: "flex",
+    justifyContent: "space-between",
+    padding: 24,
```
Comment (tdellard1) [inline client/src/components/NavBar/index.js]: Would it be possible to change some of these values up to closer match the mock up? I tried adding 24px 36px here and adding margin of 18px by margin for the logo. It looked fairly similar to the mark up nav bar

**Review by tdellard1 — APPROVED** [review]
I think since this a skeleton, it is good to merge.
